### PR TITLE
plant enable/disable sequence

### LINKED ISSUE #1386: Chiller plant enable sequence
Review chiller plant enabled/disable sequence.

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #1386.

### CODE REVIEW

**Comment by JayHuLBL** [comment]
@milicag It's ready for review.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo [hunk 1]
```diff
@@ -0,0 +1,395 @@
+﻿within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Generic;
+block PlantEnable "Sequence to enable and disable plant"
+
+  parameter Boolean haveWSE = true
+    "Flag to indicate if the plant has waterside economizer";
+  parameter Real schTab[4,2] = [0,1; 6*3600,1; 19*3600,1; 24*3600,1]
+    "Plant enabling schedule allowing operators to lock out the plant during off-hour";
+  parameter Modelica.SIunits.Temperature TChiLocOut=277.5
+    "Outdoor air lockout temperature below which the chiller plant should be disabled";
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo]: Though according to the sequence the plant can operate until oat<lockoutT - 1, which I think they should edit in the document (move that essentially hysteresis offset 1 degree up).

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+  parameter Modelica.SIunits.Time plaThrTim = 15*60
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo]: Wouldn't it be easier to use enable and disable delay, or minimum on and off time for this and the parameter below?

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+    "Threshold time to check status of chiller plant";
+  parameter Modelica.SIunits.Time reqThrTim = 3*60
+    "Threshold time to check current chiller plant request";
+  parameter Integer ignReq = 0
+    "Ignorable chiller plant requests";
+  parameter Integer iniSta = 1
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo]: Not needed, plant should always start at stage 1 if no WSE or predicted temperature not low enough (similar to my other comment, line 133).

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+  parameter Modelica.SIunits.VolumeFlowRate VHeaExcDes_flow=0.01
+    "Desing heat exchanger chilled water flow rate"
+    annotation (Dialog(group="PHXLWT", enable=haveWSE));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uPla
+    "Current chiller plant enabling status"
+    annotation (Placement(transformation(extent={{-240,200},{-200,240}}),
+      iconTransformation(extent={{-140,80},{-100,120}})));
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput chiWatSupResReq
+    "Cooling chilled water supply temperature setpoint reset request"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo]: Maybe "Number of chiller plant cooling requests" or similar.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+    annotation (Placement(transformation(extent={{200,160},{220,180}}),
+      iconTransformation(extent={{100,-10},{120,10}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Economizer.Subsequences.PredictedOutletTemperature
+    wseTOut(
+    final heaExcAppDes=heaExcAppDes,
+    final cooTowAppDes=cooTowAppDes,
+    final TOutWetDes=TOutWetDes,
+    final VHeaExcDes_flow=VHeaExcDes_flow) if haveWSE
+    "Calculated predicted heat exchanger leaving water temperature"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo]: "Waterside economizer outlet temperature predictor"

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+    "Check if enabling schedule is active"
+    annotation (Placement(transformation(extent={{-100,130},{-80,150}})));
+  Buildings.Controls.OBC.CDL.Logical.Not not1 "Logical not"
+    annotation (Placement(transformation(extent={{-140,210},{-120,230}})));
+  Buildings.Controls.OBC.CDL.Logical.Timer disTim
+    "Chiller plant disabled time"
+    annotation (Placement(transformation(extent={{-100,210},{-80,230}})));
+  Buildings.Controls.OBC.CDL.Integers.GreaterThreshold hasReq(
+    final threshold=ignReq)
+    "Check if the chiller plant request is greater than ignorable request"
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo]: Maybe add "...the number of..."

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+          extent={{-66,46},{76,10}},
+          lineColor={28,108,200},
+          textString="START"),
+        Text(
+          extent={{-66,-8},{76,-44}},
+          lineColor={28,108,200},
+          textString="STOP")}),
+ Documentation(info="<html>
+<p>
+Block that generate chiller plant enable signals and output the initial plant stage,
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo]: Maybe I should document WSE similarly, let me check in the next meeting and I can add it to this branch once this is ready.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+  Buildings.Controls.OBC.CDL.Discrete.TriggeredSampler triSam
+    "Triggered sampling of continuous signals"
+    annotation (Placement(transformation(extent={{140,-130},{160,-110}})));
+  Buildings.Controls.OBC.CDL.Logical.Edge edg
+    annotation (Placement(transformation(extent={{120,-170},{140,-150}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant staOne(final k=1)
+    "Stage one"
+    annotation (Placement(transformation(extent={{0,-150},{20,-130}})));
+  Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys(
+    final uLow=-1 + hysDt,
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo]: This should be 0

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo [hunk 9]
```diff
@@ -0,0 +1,398 @@
...
+</li>
+<li>
+Otherwise, the initial stage will be 1.
+</li>
+
+</ul>
+</li>
+</ol>
+
+
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/PlantEnable.mo]: Just noting 2 empty lines, maybe they are on purpose.

File: Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/package.mo [hunk 10]
```diff
@@ -0,0 +1,43 @@
+within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant;
+package Generic "Generic control sequences"
+
+annotation (preferredView="info", Documentation(info="<html>
+<p>
+This package contains generic sequences that would be needed for plant control.
```
Comment (milicag) [inline Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Generic/package.mo]: Maybe change to "contains generic primary plant control sequences".

**Pull request merged**